ui-tests

IvanBelov2991 opened the pull request.

IvanBelov2991 committed:
ui-tests

AlexKitIt commented on data.py:
```diff
@@ -0,0 +1,14 @@
+class Urls:
+    MAIN_PAGE = 'https://stellarburgers.nomoreparties.site/'
+    ORDERS_PAGE = 'https://stellarburgers.nomoreparties.site/feed'
+    LOGIN_PAGE = 'https://stellarburgers.nomoreparties.site/login'
+    REGISTER_PAGE = 'https://stellarburgers.nomoreparties.site/register'
+    RECOVER_PASSWORD_PAGE = 'https://stellarburgers.nomoreparties.site/forgot-password'
+    RESET_PASSWORD_PAGE = 'https://stellarburgers.nomoreparties.site/reset-password'
+    PERSONAL_ACCOUNT_PAGE = 'https://stellarburgers.nomoreparties.site/account/profile'
+    PERSONAL_ACCOUNT_HISTORY_PAGE = 'https://stellarburgers.nomoreparties.site/account/order-history'
+    ORDER_LIST_PAGE = 'https://stellarburgers.nomoreparties.site/feed'
```
Можно лучше: урлы стоит конкатенировать с базовым, так будет проще поддерживать несколько стендов

AlexKitIt commented on pages/base_page.py:
```diff
@@ -0,0 +1,103 @@
+    @allure.step('Дождаться отображения элемента')
+    def is_element_visible(self, locator):
+        try:
+            element = WebDriverWait(self.driver, 10).until(
+                expected_conditions.visibility_of_element_located(locator))
```
Нужно исправить: здесь и далее: метод описан в wait_and_find_element необходимо переиспользовать а не копипастить

AlexKitIt commented on tests/test_login_page.py:
```diff
@@ -0,0 +1,85 @@
+        login_page = LoginPage(driver)
+        login_page.open_page(Urls.LOGIN_PAGE)
+        login_page.wait_for_load_page(Urls.LOGIN_PAGE)
+        login_page.click_element(login_page.RECOVER_PASSWORD_HYPERLINK_TEXT)
```
Нужно исправить: здесь и далее: локаторов в тестах быть не должно, если это не вариант параметризации, они живут в методах пейджи, тест должен быть максимально изолирован от логики выполнения

AlexKitIt commented on tests/test_login_page.py:
```diff
@@ -0,0 +1,85 @@
+        login_page.click_element(login_page.LOGIN_BUTTON)
+        login_page.wait_for_load_page(Urls.MAIN_PAGE)
+
+        assert driver.current_url == Urls.MAIN_PAGE
```
Нужно исправить: здесь и далее: прямого обращения к driver из тестов быть не должно , только через методы пейджей

IvanBelov2991 committed:
ui-tests - fix
В частности в main_page.py изменил метод "def get_id_order", так как машина/сайт стали работать быстрее и валидный id заказа не отлавливался даже по явному ожиданию конца лоадера (ловился id 9999) - в методе добавился time.sleep, что, на первый взгляд, не очень хорошо, но он используется в контексте - "если поймал id 9999, то попробуй поймать другой id через секунду" - что, мне кажется, не так уж и плохо, так как явные ожидания не работают как надо

AlexKitIt commented on pages/main_page.py:
```diff
@@ -38,28 +51,145 @@ def get_order_count_orders_for_all_time_done(self):
+        retry_count = 0
+        max_retries = 5
+        while not is_valid_order_id(order_id) and retry_count < max_retries:
+            time.sleep(1)
```
Нужно исправить: что мешает поставить ожидание until_not 9999 в id ?
